feat : ingredient-category page

soyeooooo opened the pull request:
feat : ingredient-category page , card , feed 구현

soyeooooo committed:
feat : ingredient-category card

card(반응형) 구현

soyeooooo committed:
feat : ingredient-category page, feed

ingredient-category feed, page 구현

clianor commented:
네이밍과 같은 부분만 좀 더 신경쓰고 다시 요청해주세요!

clianor commented on src/pages/ingredient-category/index.ts:
```diff
@@ -0,0 +1 @@
+export { default as CategoryPage } from './ingredient-category-page.ui';
```
export 를 다른 파일과 동일하도록 맞춰주세요

clianor commented on src/pages/ingredient-category/ingredient-category.stories.tsx:
파일명은 항상 같은 폴더내 다른 파일들과 같게 유지해주세요.

clianor commented on src/features/ingredient-category-card/ingredient-category.contract.ts:
해당 파일은 불필요한 파일 같습니다.
그리고 contract는 유효성 검사를 위한 zod 스키마만 두려고 하는데 참고해주시면 좋을것 같아요.

clianor commented on src/pages/ingredient-category/ingredient-category-page.ui.tsx:
컴포넌트 이름은 폴더 명칭과 연관지어 작성해주세요.
카테고리 페이지라는 이름을 보고선 정확한 위치를 찾을 수 없을 것 같습니다.

clianor commented on src/widgets/category-ingredient-feed/category-ingredient-feed.ui.tsx:
피드의 경우 DailyRecipesFeed와 같이 구성해보시는건 어떨까요.
features/ingredient-category-card의 코드와 병합되어도 문제가 없을것 같습니다.
ingredient-category-card가 ui상으로 재사용이 되는 컴포넌트라면 분리도 괜찮을것 같습니다.

clianor replied:
해당 컴포넌트의 명칭이 명확하지 않은것 같습니다.
폴더명과 일치시키는게 좋을것 같습니다.
그런데 단순히 카테고리를 리스트로 보여주는 형태 같은데 feed보다는 container나 list와 같은게 어떨가요?

clianor commented on src/features/ingredient-category-card/ingredient-category-card.ui.tsx:
```diff
@@ -0,0 +1,53 @@
+      {...props}
+    >
+      {/* 이미지 영역 */}
+      <div
```
next/image 대신 backgroundImage를 이용하신 이유가 있을까요?

clianor commented on src/features/ingredient-category-card/ingredient-category-card.ui.tsx:
```diff
@@ -0,0 +1,53 @@
+      {...props}
+    >
+      {/* 이미지 영역 */}
+      <div
+        className="flex h-0 w-full items-center justify-center rounded-md bg-gray-200 bg-cover bg-center pb-[50%]"
+        style={{
+          backgroundImage: `url(${imageSrc})`,
+        }}
+      >
+        {!imageSrc && <span className="text-gray-500">Image</span>}
+      </div>
```
next/image 대신 backgroundImage를 이용하신 이유가 있을까요?

clianor reviewed (changes requested):
리뷰 내용을 반영해서 다시 작성해주실 수 있으실까요?

soyeooooo committed:
refactor : Edit review details (ingredient-category page)

1. 네이밍 구체화
2. 경로 형태 일치
3. 파일명 수정

soyeooooo commented:
넵 꼼꼼한 리뷰 감사합니다 ! 곧, 식단 카테고리 페이지도 추가해서 올려두도록 하겠습니다 !!

soyeooooo committed:
refactor : Edit review details (ingredient-category card, feed)

1. feed -> container 로 수정
2. contract schema 만 -> 기존 데이터부분 container.config로
3. card.config 이름 수정
4. card 변수명 ingredient~로 구체화
5. card.ui backgrond -> image로 수정

clianor commented on src/features/ingredient-category-card/ingredient-category-card.contract.ts:
```diff
@@ -0,0 +1,10 @@
+import { z } from 'zod';
+
+export const ingredientCategoryCardSchema = z.object({
+  ingredientImageSrc: z.string().url(),
+  ingredientIcon: z.string().optional(),
+  ingredientTitle: z.string().min(1),
+  ingredientDescription: z.string(),
+});
```
소연님 [src/features/create-recipe/create-recipe.contract.ts](https://github.com/Do-Healthy/DO-Healthy-Front/blob/develop/src/features/create-recipe/create-recipe.contract.ts) 파일을 확인해보시면 Schema가 대문자로 시작하고 있을거에요.
이 부분과 함께 맞춰주시면 좋을것 같아요.

코드 컨벤션의 경우에 맞추지 않았고 기록도 해주지 않아 매번 설명드려야하는점 죄송합니다.

clianor commented on src/features/ingredient-category-card/ingredient-category-card.contract.ts:
```diff
@@ -0,0 +1,10 @@
+  ingredientDescription: z.string(),
+});
+
+export type IngredientCategoryCardProps = z.infer<typeof ingredientCategoryCardSchema>;
```
Props에 관한 타입은 컴포넌트와 함께 응집되게 두시거나 .types.ts 파일로 분리해주시는게 좋을것 같아요.
.contract.ts 파일은 zod 스키마와 해당 스키마와 동일한 이름을 가진 타입만 생성하는 목적의 파일이라고 보시면 될것 같습니다.

soyeooooo committed:
refactor : Clean up card and container components

주석 추가, 스토리북 폴더이름 수정, 필요없는 데이터 삭제

clianor commented on src/features/ingredient-category-card/ingredient-category-card.ui.tsx:
```diff
@@ -0,0 +1,64 @@
+
+import { cn } from '@/shared/utils';
+
+interface IngredientCategoryCardProps extends React.HTMLAttributes<HTMLDivElement> {
+  ingredientImageSrc: string;
+  ingredientIcon?: React.ReactNode;
+  ingredientTitle: string;
+  ingredientDescription: string;
+}
```
React.HTMLAttributes의 경우 저도 shadcn이 제공해주는걸 따로 수정하지 않아 제 코드를 참고하면서 생긴 문제네요 이 부분은 죄송합니다.
저번에 말씀드린것처럼 ComponentProps를 이용해주시면 좋을것 같습니다.

soyeooooo committed:
feat : diet-category card, container, page

식단 카테고리 카드 컨테이너 페이지 구현

clianor commented on src/widgets/category-ingredient-container/category-ingredient-container.ui.tsx:
```diff
@@ -0,0 +1,31 @@
+
+import { IngredientCategoryCard } from '@/features/ingredient-category-card/ingredient-category-card.ui';
+
+import { INGREDIENT_CATEGORIES } from './category-ingredient-container.config';
+
+type IngredientCategory = (typeof INGREDIENT_CATEGORIES)[number];
```
해당 아키텍처의 구조에서 저수준 타입의 경우, entities 폴더 아래에 해당 도메인에 관한 정보를 넣고 type.ts 파일을 만들어 제공하면 됩니다.
해당 타입의 경우에는 src/entities/ingredient.type.ts 파일에 작성되어 제공되면 될 것으로 보여집니다.
현재 INGREDIENT_CATEGORIES에서 사용되는 타입에 대한 스키마를 작성하는 경우 src/entities/ingredient.contract.ts 파일로 작성하면 될 것으로 보여집니다.

해당 아키텍처에 관한 설명은 아래의 링크에서 가볍게 읽어보시면 도움이 많이 될것 같아요.
https://feature-sliced.design/kr/docs/get-started/overview

clianor reviewed (changes requested):
고생 많이 하셨습니다.
이번 리뷰까지는 아키텍처에 관한 이해를 위한 리뷰라고 이해해주시면 될 것 같아요.
혹시 아키텍처에서 폴더 구조에 관해 이해가 안가신다면 디스코드로 말씀주세요.

clianor commented:
소연님 작업하실때 feature 브랜치를 나눠서 작업하고 자기 레포에 푸시하신 뒤 PR을 요청해주세요.
그러면 해당 feature에서 작업한 내용과 관련된 PR에서는 이어서 개발을 진행할 수 있습니다.
